### ISSUE TITLE
Deep Learning for Coders with fastai and PyTorch

### ISSUE THREAD

**Issue by vvksh** [vvksh/learning_stuff#3]
https://learning.oreilly.com/library/view/deep-learning-for/9781492045519/

Also check out the interview of author: jeremy howard on lex fridman show

=> 
I will read the book in two passes; first scan through all chapters quickly, then will go through and do exercises.

I already read through chapter 1 sometime ago which introuces the fastai library; will start from chapter 2.

total of 19 chapters; expecting to finish first scan in a week or so, say Jan 15; will do the second pass with exercises depending on how much bandwidth I have; will probably be a long tail.

For the exercises, I am using notebooks (with gpu from paperspace.com); tried to setup google cloud notebooks with frustrating experience from unclear error about quotas and zones.

Progress:

- [x] chapter 1
- [x] chapter 2
- [x] chapter 3
- [x] chapter 4
- [x] chapter 5
- [x] chapter 6
- [x] chapter 7
- [x] chapter 8
- [x] chapter 9
- [x] chapter 10
- [x] chapter 11
- [x] chapter 12
- [x] chapter 13
- [x] chapter 14
- [x] chapter 15
- [x] chapter 16
- [x] chapter 17
- [x] chapter 18
- [x] chapter 19

**Comment by vvksh** [comment]
chapter2: end-to-end description of developing, deploying a DL solution; recommended using Viola to turn notebooks into webapps (local server) and binder to turn notebook into a deployed webapp using github.

- also touched on problem of out-of-domain data and data shift being issues when deployed in production.

**Comment by vvksh** [comment]
chaper 4: gives you a general feeling of what a nn is; some warm up using pytorch to create a simple nn; and intuition regarding how SGD works.

**Comment by vvksh** [comment]
chapter 5: practical overview of handling images for training, cross-entropy loss, picking learning rate, freezing layers, applying different learning rates, how much to train. etc.

**Comment by vvksh** [comment]
chapter 6: not much, how to forumulate single/multiple classification or regression

**Comment by vvksh** [comment]
chapter 7: label smoothing and mixup to train state of the art model

**Comment by vvksh** [comment]
chapter 8: collaborative filtering; using users and example, can use nn to discover latent factors (type of content) and biases( whether the movie is good etc)

**Comment by vvksh** [comment]
chapter 9: didnt read; intro to sklearn, traditional ML etc

**Comment by vvksh** [comment]
chapter 10: language models, RNN; didnt read

**Comment by vvksh** [comment]
chapter 11: how to write transforms
chapter 12: RNNs, LSTMs, how to do it in pytorch

**Comment by vvksh** [comment]
chapter 13: barely read; intro to what a convolution is and how to do CNN in fastai and pytorch

**Comment by vvksh** [comment]
chapter 16: SGD+optimizers+how callbacks feature in fastai helps add several ideas into the training loop elegantly.

**Comment by vvksh** [comment]
chapter 17: basics of neural network [skipped]
chapter 18: model interpretation; shows how to use pytorch's hook to implement CAM (class activation map) (lets you see which part of image lead to decision making)

**Comment by vvksh** [comment]
chapter 19: shows how to implement a fastai learner with all pieces, decent code, didnt go through it.

**Comment by vvksh** [comment]
verdict: more info on theory using fastai library as facilitator that I expected; definitely learned about dataloaders, pytorch modules, pytorch hooks etc which could be useful.
